sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -35,12 +35,15 @@ export function formatCommand(commandName: string): string {
  * If anything fails, an error is thrown.
  * @internal
  */
-async function runCommands(commands: string[][]): Promise<string[]> {
+async function runCommands(commands: string[][], timeout?: number): Promise<string[]> {
   const results: string[] = [];
 
   for (const command of commands) {
     const [file, ...parameters] = command;
-    const result = (await processUtils.execFile(file, parameters, { encoding: "utf8" })) as string;
+    const result = (await processUtils.execFile(file, parameters, {
```

Comment:
Should we default the timeout value if not given?  For example, Node defaults to 0: https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback

Reply:
We could, but I assumed the process would do that internally. Also what if, tomorrow, node decided to change that default value and if we hard-code it, won't it break? wdyt? @mpodwysocki